pandas/core/apply.py

```diff
@@ -601,34 +601,32 @@ def normalize_dictlike_arg(
             func = new_func
         return func
 
-    def _try_aggregate_string_function(self, obj, arg: str, *args, **kwargs):
+    def _apply_str(self, obj, arg: str, *args, **kwargs):
         """
         if arg is a string, then try to operate on it:
-        - try to find a function (or attribute) on ourselves
+        - try to find a function (or attribute) on obj
         - try to find a numpy function
         - raise
         """
         assert isinstance(arg, str)
 
-        f = getattr(obj, arg, None)
-        if f is not None:
+        if hasattr(obj, arg):
+            f = getattr(obj, arg)
```

Comment:
Not sure I see the value of this change (but really, not much opposition to it either). My understanding of `hasattr` is that it is essentially:

```
try:
    getattr(obj, arg)
    return True
except AttributeError:
    return False
```

Now we're calling getattr twice whereas before it was only once.

Reply:
Ok, I'm mostly thinking about code organization (lining up the if/elif/else), though if there is a performance difference, it looks like it's max around some 10s of nanoseconds, i.e. negligible.

Reply:
> Ok, I'm mostly thinking about code organization (lining up the if/elif/else)

Early return is also a viable strategy:

```
def foo(x):
    if x == 2:
        return 3
    ... rest of method ...
```

as opposed to 

```
def foo(x):
    if x == 2:
        return 3
    else:
        ... rest of method ...
```

Early return keeps the indentation level lower and can make complex methods more readable.

That isn't to say I'm opposed to the changes here, but rather it seems to me someone can come along next week and undo this and I wouldn't be opposed to that either. Similar with the getattr/hasattr.

Reply:
Ok, preferences can differ, no big deal. If I revert the code inside the method, would the PR then be ok?

Reply:
Yes - and if you felt strongly about these changes I wouldn't object.

Reply:
Ok, if its not a big issue wuth you eithrer, I would prefer then to get it in as-is, just to save the time rewriting...